packages/ui/src/machines/actors/auth/signIn.ts

```diff
@@ -401,6 +451,11 @@ export const signInActor = createMachine<SignInContext, AuthEvent>(
 
         return result;
       },
+      async validateFields(context, _event) {
+        const { formValues } = context;
+        const validators = [passwordMatches]; // this can contain custom validators too
+        return runValidators(formValues, validators);
+      },
```

Comment:
Can we share this in `../../actions`?

Reply:
I wonder how that works with services? I'll take a look at this for a future PR!

Reply:
Oh shoot, yeah this is services 😓  my bad. We should be able to share those anyways, but I agree and this is not imminent.